### PR TITLE
Updated llm_tracker to support new event types

### PR DESCRIPTION
**Pull request by HowieG**
## 📥 Pull Request

**📘 Description**
Updated llm_tracker to support new event types

**🔄 Related Issue (if applicable)**
ENG-95

**🎯 Goal**
Fix llm_tracker

**🔍 Additional Context**
Modified ErrorEvent and LLMEvent schemas in supabase

**🧪 Testing**
Run tests/openai_handlers/_test_handler_openai_v0.py
Run tests/openai_handlers/_test_handler_openai_v1.py

Thank you for your contribution to Agentops!

### CODE REVIEW

**Comment by HowieG** [comment]
Purposefully left comments in so that I remember to create new Linear issues or so that we can discuss. Will remove them in followup commit

File: agentops/llm_tracker.py [hunk 1]
```diff
@@ -61,8 +58,9 @@ def handle_stream_chunk(chunk):
                     self.client.record(self.event_stream)
                     self.event_stream = None
             except Exception as e:
+                # TODO: This error is specific to only one path of failure. Should be more generic or have different logging for different paths
```
Comment (braelyn-ai) [inline agentops/llm_tracker.py:65]: 👍

**Review by braelyn-ai — APPROVED** [review]

**Review by braelyn-ai — APPROVED** [review]

File: agentops/client.py [hunk 2]
```diff
@@ -152,31 +150,27 @@ def _record_event_sync(self, func, event_name, tags, *args, **kwargs):
...
-            # Record the event after the function call
-            self.record(Event(event_type=event_name,
-                              params=arg_values,
-                              returns=returns,
-                              result='Success',
-                              action_type='action',
-                              init_timestamp=init_time,
-                              tags=tags))
+            event.returns = returns
+            # TODO: If func excepts this will never get called
```
Comment (siyangqiu) [inline agentops/client.py]: What about end time stamp? IIRC, that gets stamped at when the event gets created. You need to either update it here or create the event here (and creating the init timestamp where the event right now)

File: agentops/config.py [hunk 3]
```diff
@@ -20,7 +21,7 @@ class Configuration:
     """
 
     def __init__(self, api_key: str,
-                 endpoint: Optional[str] = 'https://api.agentops.ai',
+                 endpoint: Optional[str] = environ.get('AGENTOPS_API_ENDPOINT', 'https://api.agentops.ai'),
```
Comment (siyangqiu) [inline agentops/config.py:24]: Nice!

File: agentops/event.py [hunk 4]
```diff
@@ -9,15 +9,18 @@
...
-from uuid import UUID
+from uuid import UUID, uuid4
 
 
 @dataclass
 class Event:
     event_type: str  # EventType.ENUM.value
     tags: Optional[List[str]] = None
+    params: Optional[str] = None
+    returns: Optional[str] = None
```
Comment (siyangqiu) [inline agentops/event.py:19]: I thought LLMEvent shouldn't have returns?

File: agentops/client.py [hunk 5]
```diff
@@ -194,36 +188,28 @@ async def _record_event_async(self, func, event_name, tags, *args, **kwargs):
...
-            # Record the event after the function call
-            self.record(Event(event_type=event_name,
-                              params=arg_values,
-                              returns={f"{type(e).__name__}": str(e)},
-                              result='Fail',
-                              action_type='action',
-                              agent_id=agent_id,
-                              init_timestamp=init_time,
-                              tags=tags))
+            self.record(Error(event=event, details={f"{type(e).__name__}": str(e)}))
```
Comment (siyangqiu) [inline agentops/client.py]: Can you leave a TODO to add the stack trace?

**Review by siyangqiu — APPROVED** [review]
Looks good, except for the end_timestamp thing (unless I'm wrong).

**Pull request merged**